packages/react-heartwood-components/src/components/Core/components/HeaderPrimary/HeaderPrimary.js

```diff
@@ -83,7 +83,7 @@ export default class HeaderPrimary extends Component<Props, State> {
 		if (
 			e.key === 'Escape' ||
 			e.target.contains(this.ref) ||
-			(e.type === 'click' && e.target.id !== 'avatar')
+			(e.type === 'click' && !e.target.closest('#userMenuToggle'))
```

Comment:
Couple questions...
1. Is there ever a scenario where we'd have multiple `UserMenu` components on a page?
2. Is there a more React-y way of doing this w/ refs instead of looking at the DOM element directly?

Reply:
It's hard to imagine a scenario where there are multiple UserMenu components on the page, but not knowing the answer to that question, I agree using a ref is more suitable than an id.